### ISSUE TITLE
HLS support

### ISSUE THREAD

**Issue by i8beef** [netsmarttech/node-red-contrib-ui-media#11]
May be out of scope, but figured I'd ask as it'd let me drop a template node or two... Any chance of supporting HLS streams with something like hls.js? Im not sure what the options would be for dynamically loading the hls.js library on demand if someone for instance specified an HLS m3u8 playlist file...

Example: https://hlsbook.net/playing-hls-video-in-the-browser/

**Comment by gfcittolin** [comment]
You can display live streams using this node by sending a message with the `src` property set to the target streaming URL. We've used this to display live streams of IP cameras without a problem.
Would this be enough, or do you need some other functionality?

**Comment by i8beef** [comment]
No, because HLS isn't a natively supported HTML5 video type, but CAN be played in HTML5 video elements by using the media extensions stuff... it just requires loading the the aforementioned JS to do it (i.e., its not a browser plugin or anything). Chromecast requires HLS streams so its common for video camera streams (the other being RTSP/RTMP, but those (a) aren't Chromecast compatible, and (b) can't be put in players like this without plugins).

Essentially, it'd be cool if there was an additional node for HLS streams that could load that JS automatically. Right now I just do it with a template node manually, but having a unified "media player" project like this that could do it would be cool. I'd say separate node because there's no reason to load the HLS stuff unless you are actually displaying HLS content and that would give you a nice separation both code-wise and dependency-wise while presenting a united front on video media player capabilities for node-red.

Or it could be out of scope for you. What I have works, I would just consider switching to your node if it did this. I could provide the code from my template node if you wanted a head start on it though.

**Comment by gfcittolin** [comment]
I like the idea, that was somewhat the goal of this node - to enable the dashboard to easily reproduce any media, even though this is not 100% fulfilled today. It could also load the HLS library dynamically, depending on the type of media being displayed.

The only issue is we currently don't have time/people to allocate for this development and testing, but we'd surely accept a pull request for it, in the case you (or anyone else) could develop this.

**Comment by vkuehn** [comment]
sound great, can you kindly mention more details on how to work with ip cameras / webstreams in your Readme.md ?
